Automattic/wp-calypso
# Implement tracking via GTM for woo trials upsells to ecommerce

Related to #

## Proposed Changes
This is still a WIP, but requesting an early review, to make sure we're on the right path. 

* Pass the SiteDetails object to tracking library, and use this to track users who currently have a Woo trial plan, and are upgrading to a paid ecommerce plan.
* Implement the Google Tag Manager pixel for Woo in certain scenarios, e.g. if a user upgrades from a Woo Express trial plan to a paid ecommerce plan
* Push a purchase event with meta data to the data layer.

This is just for testing for now, to see how to best implement this.

## Testing Instructions



* Need to add automated tests
* Sandbox store, allow ad-tracking flag.
* Upgrade a Woo Express trial site via the CTA in the /plans page
* When completing the payment, observe that the GTM container is added, and that the conversion data is pushed to the data layer for GTM ID `GTM-W64W8Q`

## Pre-merge Checklist
